plugins/utils/makeFeedHtml.js

```diff
@@ -0,0 +1,53 @@
+const { imageWrapperClass } = require('gatsby-remark-images/constants')
+const unified = require('unified')
+const { convertHastToHtml } = require('./convertHast.js')
+const { select, selectAll } = require('hast-util-select')
+
+const rootToAbsolute = siteUrl => tree => {
+  selectAll('a', tree).forEach(node => {
+    if (node.properties.href.startsWith('/')) {
+      node.properties.href = siteUrl + node.properties.href
+    }
+  })
+  selectAll('img', tree).forEach(node => {
+    if (node.properties.src.startsWith('/')) {
+      node.properties.src = siteUrl + node.properties.src
+    }
+  })
+  return tree
+}
+
+const unwrapImages = () => tree => {
+  selectAll(`.${imageWrapperClass}`, tree).forEach(node => {
+    // Set the fallback image as the wrapper's only child, and then
+    // give that image the wrapper's original style.
+    const fallbackImg = select('img', node)
+    node.children = [
+      {
+        ...fallbackImg,
+        properties: {
+          ...fallbackImg.properties,
+          style: 'max-width: 100%; margin: auto;'
+        }
+      }
+    ]
+  })
+  return tree
+}
+
+function makeFeedHtml(htmlAst, siteUrl) {
```

Comment:
really annoying that gastby plugin feed does not handle this ... kinda feels like all of this is a requirement for that plugin to be useful

Reply:
I agree to a certain extent, especially for our use case, but RSS feeds are very often used for something like podcasts where the encoded content is more of a summary than the whole blog (ads are a lot harder in an RSS reader, after all), and the encoded content is kept simple.

This isn't to say the default and suggested queries of `gatsby-plugin-feed` are good, I think naively passing through the content's raw `html` string is kind of poor way to go about things, but it's not as if the plugin is useless without this kind of content reprocessing.